extensions/pkg/webhook/controlplane/controlplane.go

```diff
@@ -44,6 +44,8 @@ type Args struct {
 	Types []extensionswebhook.Type
 	// Mutator is a mutator to be used by the admission handler.
 	Mutator extensionswebhook.Mutator
+	// ObjectSelector is the label query for resources
+	ObjectSelector *metav1.LabelSelector
```

Comment:
There are also other webhooks in the extension library:
- cloudprovider-webhook: https://github.com/gardener/gardener/blob/10d75787a132bf39c408b74372f5d8c045fa8f4b/extensions/pkg/webhook/cloudprovider/cloudprovider.go#L26-L30: @kon-angelo do you recall why the object selector here is configurable and not always enforced?
- network- webhook: https://github.com/gardener/gardener/blob/10d75787a132bf39c408b74372f5d8c045fa8f4b/extensions/pkg/webhook/network/network.go#L24-L34
- shoot-webhook: https://github.com/gardener/gardener/blob/10d75787a132bf39c408b74372f5d8c045fa8f4b/extensions/pkg/webhook/shoot/shoot.go#L29-L39

It make sense to add in a similar way object selectors to these webhooks as well, right? But let's better do this in follow-up PRs to do not block this PR.

Reply:
After offline discussion with @LucaBernstein : there is no need of object selector in the network webhook. There is only one Network resource in the Shoot control plane namespace in the Seed.
We still need to sort out why the object selector is not enforced in the cloudprovider-webhook and we need to make the object selector configurable for the shoot-webhook. But these items will be tracked separately in https://github.com/gardener/gardener/issues/9864.